content/docs/5.1/concepts/explore-tikv-features/cas.md

```diff
@@ -0,0 +1,137 @@
+---
+title: CAS on RawKV
+description: Compare And Swap
+menu:
+    "5.1":
+        parent: Features
+        weight: 4
+---
+
+This page walks you through a simple demonstration of how to use TiKV's `compare-and-swap`.
+
+In RawKV, `compare-and-swap (CAS)` is an atomic instruction used in multi-threading to achieve synchronization. It compares the `current_value` of a `key` with a given value and, only if they are the same, modifies the contents of that `key` to a new given value. This is done as a single atomic operation. The atomicity guarantees that the new value is calculated based on up-to-date information; if the value had been updated by another thread in the meantime, the write would fail.
+
+## Prerequisites
+
+1. Install TiUP by executing the following command:
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | sh
+```
+
+2. If TiUP is already installed, update the TiUP playground component to the latest version:
+
+```bash
+tiup update --self && tiup update playground
+```
+
+3. Install Rust
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+## Step 1: Start TiKV Cluster
+
+For the purpose of this tutorial, you need only one TiKV node, so use the `tiup playground` command.
+
+Show TiUP version:
+
+```bash
+tiup -v
+```
+
+version >= 1.5.0:
+
+```bash
+tiup playground --mode tikv-slim
+```
+
+version < 1.5.0:
+
+```bash
+tiup playground
+```
+
+## Step 2: Create a rust project
+
+Use `cargo` to create a rust project:
+
+```bash
+cargo new hello-tikv
+cd hello-tikv
+```
+
+## Step 3: Add dependency
+
+Append the following content to file `Cargo.toml`.
+
+```toml
+tikv-client = "0.1.0"
+tokio = { version = "1.5.0", features = ["full"] }
+```
+
+## Step 4: Write test code
```

Comment:
maybe it's better to use Java as the first language for examples since it's most widely used in application developments.

Reply:
As discussed, RawKV's examples will use `java`. 
So I will change the example of `CAS` from `rust` to `java`.

Reply:
Can we continue to review the PR? I will update the example as soon as the CAS API is ready in java client.

Reply:
OK.